pilot/pkg/controllers/ipallocate/ipallocate.go

```diff
@@ -219,7 +221,11 @@ func (c *IPAllocator) resolveConflict(conflict conflictDetectedEvent) error {
 	var serviceentries, autoConflicts, userConflicts []*networkingv1.ServiceEntry
 
 	for _, conflictingAddress := range conflict.getAddresses() {
-		serviceentries = append(serviceentries, c.index.Lookup(conflictingAddress)...)
+		for _, potentialentries := range c.index.Lookup(conflictingAddress) {
+			if c.getObjectFilter() == nil || c.getObjectFilter().Filter(potentialentries) {
```

Comment:
Isn't the filter has been paseed to the informer? Why we need to do the filtering again in the upper caller

Reply:
https://github.com/istio/istio/issues/54300 explains it. Seems like maybe Indexers have a bug and this is working correcting for that. Maybe I misunderstand and it's meant to ignore the filtering but I can't see why straight away anyhow.

Reply:
@therealmitchconnors, just brainstorming here.  Could we filter inside the `CreateIndex` extractor so that svcentry that are outside our domain have no addresses? It's probably easier to strip out later if we just check the entry and bail out early with an empty list inside that extractor.

Reply:
(we might want to strip it out if indexing get's an update to obey the filtering)